GeoDaCenter/chicago-environment-explorer
# feat: added Point data + Legend for Cooling Centers

## Problem
We would like to add a layer that will display Point data showing all the different types of Cooling Centers in the city

For reference: https://data.cityofchicago.org/Health-Human-Services/Cooling-Centers-Map/cj7n-sh49

Fixes #146 

## Approach
* feat: scraped this public API for JSON data, converted to GeoJSON, and added to `public` resources
* feat: added a new layer (overlay) to display this Point data - ~~uses existing colors from the link above~~
* feat: added a Legend for new Cooling Centers overlay - ~~uses existing colors from the link above~~

### TODOs

* ~~allow user to select multiple data overlays at once~~ done - we may need to adjust or allow user to choose the order that overlay layers are rendered
* ~~adjust colors to avoid overlap with variable layer(s)~~ done - now uses various shades of blue
* ~~display detailed info when point is clicked: address, phone number, hours of operation, etc~~ done

## How to Test
1. Navigate to https://deploy-preview-159--chicago-env-explorer.netlify.app/map
2. Scroll down to the Data Overlays dropdown and expand it
    * You should see the new Cooling Centers overlay is listed here
3. Select the Cooling Centers overlay
    * You should see color-coded point data appear on the map
    * You should see a Legend appear below the dropdown showing the meaning of each colored point
    * You should see that this overlay is now selected **in addition to** the Community Areas overlay
4. Hover over a Cooling Center map point with the mouse cursor
    * You should see a tooltip appear that contains site_type, site_name, address, phone number (if applicable), and hours of operation (if applicable)
5. Compare with https://data.cityofchicago.org/Health-Human-Services/Cooling-Centers-Map/cj7n-sh49


![Screenshot 2024-05-24 at 1 41 53 PM](https://github.com/GeoDaCenter/chicago-environment-explorer/assets/1413653/a3e82127-dc68-4523-970e-099bf7c8cf55)
